### PR TITLE
Kik 8090 Check balance before sending transactions and better error messages

### PR DESCRIPTION
**Pull request by natanrolnik**

### CODE REVIEW

File: KinSDK/KinSDK/KinError.swift [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+        case .unsupportedNetwork:
+            return "Unsupported Network"
+        case .invalidAddress:
+            return "Invalid Address"
+        case .invalidAmount:
+            return "Invalid Amount"
+        case .insufficientBalance:
+            return "Not enough Kin"
+        default:
+            return "Unknown error"
```
Comment (avi-kik) [inline KinSDK/KinSDK/KinError.swift]: Better to cover every case and avoid the default.  The compiler will then ensure that new errors are localized.

File: KinSDK/KinSDK/KinAccount.swift [hunk 2]
```diff
@@ -63,7 +63,13 @@ public class KinAccount {
             let price = try? store.client.suggestGasPrice(store.context),
             let toAddress = GethNewInterface(),
             let value = GethNewInterface() else {
-                throw KinError.internalInconsistancy
+                throw KinError.internalInconsistency
+        }
+
+        let currentBalance = (try balance() as NSDecimalNumber).uint64Value
```
Comment (YohayBar) [inline KinSDK/KinSDK/KinAccount.swift]: what happen if try fails?

File: KinSDK/KinSDK/KinError.swift (hunk 1 above)
Comment (natanrolnik) [reply KinSDK/KinSDK/KinError.swift]: Yeah, you are right. I'll fix it.

File: KinSDK/KinSDK/KinAccount.swift (hunk 2 above)
Comment (natanrolnik) [reply KinSDK/KinSDK/KinAccount.swift]: the whole method `throws`, so if `balance()` fails, sending a transaction will fail as well.

Comment (YohayBar) [reply KinSDK/KinSDK/KinAccount.swift]: 👍

**Pull request merged**